dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Relation.java

```diff
@@ -14,12 +15,37 @@
 public final class Relation implements Dependent<Relation> {
 
     private final Wmm wmm;
+    private final boolean unary;
     Definition definition = new Definition.Undefined(this);
     private boolean isRecursive;
     final List<String> names = new ArrayList<>();
 
-    Relation(Wmm wmm) {
+    Relation(Wmm wmm, boolean unary) {
```

Comment:
Not a big fan of the boolean value here. I think an arity integer or even better, an arity enum (Arity.UNARY, Arity.BINARY), is the better choice.